parl/remote/worker_manager.py

```diff
@@ -0,0 +1,235 @@
+#   Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import threading
+import random
+from collections import defaultdict
+from collections import namedtuple
+from parl.utils import logger
+from parl.remote import remote_constants
+from parl.remote.cpu_resource import CpuResource
+from parl.remote.gpu_resource import GpuResource
+from parl.remote.message import InitializedCpu
+from parl.remote.message import InitializedGpu
+
+
+class WorkerManager(object):
+    """The WorkerManager deals with everything related to workers.
```

Comment:
A thread-safe data structure used to respond to resource request, maintaining the available computation resources in each worker.